ctru-rs/src/services/am.rs

```diff
@@ -56,6 +68,55 @@ impl<'a> Title<'a> {
             Ok(info.assume_init())
         }
     }
+
+    pub fn get_smdh(&self) -> crate::Result<Smdh> {
+        // i have no idea how to make this look better
+        let archive_path_data: [u32; 4] =
+            [self.low_u32(), self.high_u32(), self.mediatype as u32, 0x0];
+        let smdh_path_data: [u32; 5] = [0x0, 0x0, 0x2, u32::from_le_bytes(*b"icon"), 0x0];
+
+        let archive_path = ctru_sys::FS_Path {
+            type_: ctru_sys::PATH_BINARY,
+            size: size_of::<[u32; 4]>() as u32,
+            data: archive_path_data.as_ptr() as *const libc::c_void,
```

Comment:
Maybe not as relevant with the other changes I mentioned, but in general:

```suggestion
            data: archive_path_data.as_ptr().cast(),
```